Doesn't compile with async_std (requires tokio)

```
retry_fn = { version = "0.2.0", features = ["async-runtime"] }
```



```
error[E0433]: failed to resolve: use of undeclared crate or module `tokio`
   --> /home/tom/.cargo/registry/src/index.crates.io-6f17d22bba15001f/retry_fn-0.2.0/src/macros.rs:130:25
    |
130 | ...             tokio::time::sleep(dur).a...
    |                 ^^^^^ use of undeclared crate or module `tokio`
```

```rs
use retry_fn::async_std::{retry, strategy::Constant, RetryResult};

#[async_std::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
    retry(Constant::from_millis(500), |op| {
        let page = page.clone();
        async move {
            if op.retries >= 3 {
                RetryResult::<&str, _>::Err(io::Error::new(io::ErrorKind::TimedOut, "timed out"))
            } else {
                page.find_xpath("//a[contains(., 'Settings/Registration')]")
                    .await?
                    .click()
                    .await?;
                RetryResult::Retry()
            }
        }
    });
...
}
```

Looks like feature flags are needed in macros.rs:

~ Line 127:
```rs
            for dur in iter.into_iter() {
                match (&mut f).await {
                    RetryResult::Retry() => {
                        // should use async_std for async_std and tokio for tokio
                        tokio::time::sleep(dur).await;
                        total_delay += dur;
                        count += 1;
                    }
                    RetryResult::Err(err) => {

```

--- a/.github/workflows/actions.yml
+++ b/.github/workflows/actions.yml
@@ -73,7 +73,6 @@ jobs:
         rust:
           - stable
           - beta
-          - nightly
     steps:
       - name: Checkout sources
         uses: actions/checkout@v2

--- a/.github/workflows/actions.yml
+++ b/.github/workflows/actions.yml
@@ -26,18 +26,9 @@ jobs:
 
       - name: cargo check
         uses: actions-rs/cargo@v1
-        with:
-          command: check
-
-      - name: cargo check tokio
-        uses: actions-rs/cargo@v1
-        with:
-          command: check --features tokio-runtime
-
-      - name: cargo check async-std
-        uses: actions-rs/cargo@v1
-        with:
-          command: check --features async-runtime
+      - run: cargo check
+      - run: cargo check --features tokio-runtime
+      - run: cargo check --features async-runtime
 
   test:
     name: Test Suite

--- a/.github/workflows/actions.yml
+++ b/.github/workflows/actions.yml
@@ -24,10 +24,20 @@ jobs:
           toolchain: ${{ matrix.rust }}
           override: true
 
-      - name: Run cargo check
+      - name: cargo check
         uses: actions-rs/cargo@v1
         with:
           command: check
+
+      - name: cargo check tokio
+        uses: actions-rs/cargo@v1
+        with:
+          command: check --features tokio-runtime
+
+      - name: cargo check async-std
+        uses: actions-rs/cargo@v1
+        with:
+          command: check --features async-runtime
 
   test:
     name: Test Suite

--- a/.github/workflows/actions.yml
+++ b/.github/workflows/actions.yml
@@ -26,9 +26,20 @@ jobs:
 
       - name: cargo check
         uses: actions-rs/cargo@v1
-      - run: cargo check
-      - run: cargo check --features tokio-runtime
-      - run: cargo check --features async-runtime
+        with:
+          command: check
+
+      - name: cargo check tokio
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --features tokio-runtime
+
+      - name: cargo check async-std
+        uses: actions-rs/cargo@v1
+        with:
+          command: check
+          args: --features async-runtime
 
   test:
     name: Test Suite

--- a/README.md
+++ b/README.md
@@ -12,7 +12,7 @@ retry libraries out there, the desire is to keep this up-to-date and combine fea
 
 ```rust
 use std::{io, time::Duration};
-use retry_fn::{retry, RetryResult, strategy::ExponentialBackoff};
+use retry_fn::{retry, strategy::ExponentialBackoff, RetryResult};
 
 fn main() -> Result<(), Box<dyn std::error::Error>> {
     let mut count = 0;
@@ -39,7 +39,7 @@ Enable the `tokio-runtime` feature to get access to this function
 
 ```rust
 use std::{io, sync::{Arc, Mutex}};
-use retry_fn::{tokio::retry, RetryResult, strategy::Constant};
+use retry_fn::{tokio::retry, strategy::Constant, RetryResult};
 
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {

--- a/rustfmt.toml
+++ b/rustfmt.toml
@@ -1,5 +1,5 @@
 unstable_features = true
-merge_imports = true
+imports_granularity = "Crate"
 merge_derives = true
 reorder_impl_items = true
 use_field_init_shorthand = true

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -9,7 +9,7 @@
 //!
 //! ```rust,no_run
 //! # use std::{io, time::Duration};
-//! use retry_fn::{retry, RetryResult, strategy::ExponentialBackoff};
+//! use retry_fn::{retry, strategy::ExponentialBackoff, RetryResult};
 //! # fn main() -> Result<(), Box<dyn std::error::Error>> {
 //! let mut count = 0;
 //! let res = retry(ExponentialBackoff::new(Duration::from_secs(2)), |op| {
@@ -167,7 +167,7 @@ where
 ///
 /// ```rust,no_run
 /// # use std::{io, time::Duration};
-/// use retry_fn::{retry, RetryResult, strategy::ExponentialBackoff};
+/// use retry_fn::{retry, strategy::ExponentialBackoff, RetryResult};
 /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
 /// let mut count = 0;
 /// let res = retry(ExponentialBackoff::new(Duration::from_secs(2)), |op| {

--- a/src/macros.rs
+++ b/src/macros.rs
@@ -127,7 +127,10 @@ macro_rules! retry_impl {
             for dur in iter.into_iter() {
                 match (&mut f).await {
                     RetryResult::Retry() => {
+                        #[cfg(feature = "tokio-runtime")]
                         tokio::time::sleep(dur).await;
+                        #[cfg(not(feature = "tokio-runtime"))]
+                        async_std::task::sleep(dur).await;
                         total_delay += dur;
                         count += 1;
                     }

--- a/src/macros.rs
+++ b/src/macros.rs
@@ -103,8 +103,7 @@ macro_rules! retry_impl {
         ///     }
         /// };
         /// tokio::pin!(fut);
-        /// let res = retry_unpin(Constant::from_millis(100), fut)
-        /// .await;
+        /// let res = retry_unpin(Constant::from_millis(100), fut).await;
         /// assert_eq!(*count.lock().unwrap(), 3);
         /// assert!(res.is_err());
         /// # });

--- a/src/macros.rs
+++ b/src/macros.rs
@@ -8,8 +8,7 @@ macro_rules! retry_impl {
         ///
         /// ```rust,no_run
         /// # use std::{io, sync::{Arc, Mutex}};
-        /// use retry_fn::strategy::Constant;
-        /// use retry_fn::RetryResult;
+        /// use retry_fn::{strategy::Constant, RetryResult};
         /// # use retry_fn::tokio::retry;
         /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
         /// # tokio::task::spawn_blocking(|| async move {
@@ -84,8 +83,7 @@ macro_rules! retry_impl {
         ///
         /// ```rust,no_run
         /// # use std::{io, sync::{Arc, Mutex}};
-        /// use retry_fn::strategy::Constant;
-        /// use retry_fn::RetryResult;
+        /// use retry_fn::{strategy::Constant, RetryResult};
         /// # use retry_fn::tokio::retry_unpin;
         /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
         /// # tokio::task::spawn_blocking(|| async move {

--- a/src/tokio.rs
+++ b/src/tokio.rs
@@ -4,7 +4,7 @@
 //!
 //! ```rust,no_run
 //! # use std::{io, sync::{Arc, Mutex}};
-//! use retry_fn::{tokio::retry, RetryResult, strategy::Constant};
+//! use retry_fn::{tokio::retry, strategy::Constant, RetryResult};
 //! # fn main() -> Result<(), Box<dyn std::error::Error>> {
 //! # tokio::task::spawn_blocking(|| async move {
 //! let count: Arc<Mutex<i32>> = Arc::new(Mutex::new(0));

